moveit_common/CMakeLists.txt

```diff
@@ -3,6 +3,16 @@ project(moveit_common NONE)
 
 find_package(ament_cmake REQUIRED)
 
+if(BUILD_TESTING)
```

Comment:
Can't we simply set and enable all tests in `moveit_common` only, possibly providing an extra macro? That way we don't have linters enabled all over the place and we can enforce what checks need to pass in all MoveIt packages. Also, what do you think about using a different build flag than `BUILD_TESTING` so that we can separate the CI stages?

Reply:
Using the `colcon mixin` system when linters are enabled this way there is an explicit way to run only linters or only non-linters.

Here is the documentation for how to enable colcon mixin: https://github.com/colcon/colcon-mixin-repository

When running tests with colon you can then only linters:
```bash
colcon test --mixin linters-only
```

Skip running linters:
```bash
colcon test --mixin linters-skip
```

Reply:
> Can't we simply set and enable all tests in `moveit_common` only

I tried this and I couldn't figure out how to get them to propagate.  I really wish there was a way to specify linters per-repo instead of per-ros-package.  I asked about this here: https://github.com/colcon/colcon-core/issues/421 to see if I could instead find some other explicit per-repo way to specify them.

The other thing is that even if we figured out how to enable them for all packages from moveit_common, until all packages pass these linters we need a way to exclude them per-package.